pgrx/src/datum/array.rs

```diff
@@ -585,6 +561,8 @@ impl<'a, T: FromDatum> Iterator for ArrayTypedIterator<'a, T> {
         if *curr >= array.raw.len() {
             None
         } else {
+            // Make sure a previous iter didn't push out of bounds!
+            debug_assert!(*ptr < array.raw.end_ptr());
```

Comment:
This might be the "couple of other places" I am thinking of. I knew I had seen it >_>